RecoLocalCalo/HcalRecProducers/src/FacileHcalReconstructor.cc

```diff
@@ -0,0 +1,101 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonClient.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicEDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+#include "Geometry/Records/interface/HcalRecNumberingRecord.h"
+
+class FacileHcalReconstructor : public SonicEDProducer<TritonClient> {
+public:
+  explicit FacileHcalReconstructor(edm::ParameterSet const& cfg)
+      : SonicEDProducer<TritonClient>(cfg),
+        fChannelInfoName_(cfg.getParameter<edm::InputTag>("ChannelInfoName")),
+        fTokChannelInfo_(consumes<HBHEChannelInfoCollection>(fChannelInfoName_)),
+        htopoToken_(esConsumes<HcalTopology, HcalRecNumberingRecord>()) {
+    produces<HBHERecHitCollection>();
+    setDebugName("FacileHcalReconstructor");
+  }
+
+  void acquire(edm::Event const& iEvent, edm::EventSetup const& iSetup, Input& iInput) override {
+    edm::Handle<HBHEChannelInfoCollection> hChannelInfo;
+    iEvent.getByToken(fTokChannelInfo_, hChannelInfo);
+
+    const HcalTopology* htopo = &iSetup.getData(htopoToken_);
+
+    auto& input1 = iInput.begin()->second;
+    auto data1 = std::make_shared<TritonInput<float>>();
+    data1->reserve(hChannelInfo->size());
+    client_.setBatchSize(hChannelInfo->size());
+
+    hcalIds_.clear();
+
+    for (const auto& pChannel : *hChannelInfo) {
+      std::vector<float> input;
+      const HcalDetId pDetId = pChannel.id();
+      hcalIds_.push_back(pDetId);
+
+      //FACILE uses iphi as a continuous variable
+      input.push_back((float)pDetId.iphi());
```

Comment:
the explicit float casts may not be necessary here (and the few lines below)
but if they are, C++ casts should be used, i.e. `static_cast<float>(x)`